File: connectors/connector-trello/src/main/scala/com/taskadapter/connector/trello/TrelloField.scala
package com.taskadapter.connector.trello

import com.taskadapter.model._

object TrelloField {

  val listId = Field("List Id")
  val listName = TaskStatus

  val fields = List(listName, listId,
    Id,
    Key,
    Summary, ReporterFullName, ReporterLoginName, Description, DueDate, UpdatedOn
  )

  val excludeFromNewConfig = Seq(Id, Key, listId, ReporterFullName, ReporterLoginName, UpdatedOn)

  val defaultFieldsForNewConfig = fields.diff(excludeFromNewConfig)
}

File: connectors/connector-trello/src/main/java/com/taskadapter/connector/trello/TrelloField.java
package com.taskadapter.connector.trello;

import com.taskadapter.model.AllFields;
import com.taskadapter.model.CustomString;
import com.taskadapter.model.Field;

import java.util.ArrayList;
import java.util.List;

public class TrelloField {
    public static final Field<String> listId = new CustomString("List Id");
    public static final Field<String> listName = AllFields.taskStatus;

    public static List<Field<?>> fields = List.of(listName, listId,
            AllFields.id,
            AllFields.key,
            AllFields.summary,
            AllFields.reporterFullName,
            AllFields.reporterLoginName,
            AllFields.description,
            AllFields.dueDate,
            AllFields.updatedOn
    );

    public static List<Field<?>> excludeFromNewConfig = List.of(
            AllFields.id,
            AllFields.key,
            listId,
            AllFields.reporterFullName,
            AllFields.reporterLoginName,
            AllFields.updatedOn);

    public static List<Field<?>> defaultFieldsForNewConfig() {
        var fieldsCopy = new ArrayList<>(fields);
        fieldsCopy.removeAll(excludeFromNewConfig);
        return fieldsCopy;
    }
}
